### ISSUE TITLE
Successful migration but broken frontend

### ISSUE THREAD

**Issue by onepack** [magento/data-migration-tool#129]
After a successful data migration the frontend has become broken with the following error:

1 exception(s):
Exception #0 (Exception): Warning: preg_match(): Delimiter must not be alphanumeric or backslash in /apps/web/dev3.site.com/public_html/vendor/magento/framework/View/DesignExceptions.php on line 70

Exception #0 (Exception): Warning: preg_match(): Delimiter must not be alphanumeric or backslash in /apps/web/dev3.site.com/public_html/vendor/magento/framework/View/DesignExceptions.php on line 70
#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2, 'preg_match(): D...', '/apps/web/dev3....', 70, Array)
#1 /apps/web/dev3.site.com/public_html/vendor/magento/framework/View/DesignExceptions.php(70): preg_match('Androidd|iPhone...', 'Mozilla/5.0 (Wi...')
#2 /apps/web/dev3.site.nl/public_html/vendor/magento/module-page-cache/Model/App/CacheIdentifierPlugin.php(43): Magento\Framework\View\DesignExceptions->getThemeByRequest(Object(Magento\Framework\App\Request\Http))
#3 /apps/web/dev3.site.com/public_html/vendor/magento/framework/Interception/Interceptor.php(152): Magento\PageCache\Model\App\CacheIdentifierPlugin->afterGetValue(Object(Magento\Framework\App\PageCache\Identifier\Interceptor), '25076b3791ece09...')
#4 /apps/web/dev3.site.com/public_html/var/generation/Magento/Framework/App/PageCache/Identifier/Interceptor.php(26): Magento\Framework\App\PageCache\Identifier\Interceptor->___callPlugins('getValue', Array, Array)
#5 /apps/web/dev3.site.com/public_html/vendor/magento/framework/App/PageCache/Kernel.php(60): Magento\Framework\App\PageCache\Identifier\Interceptor->getValue()
#6 /apps/web/dev3.site.com/public_html/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(71): Magento\Framework\App\PageCache\Kernel->load()
#7 /apps/web/dev3.site.com/public_html/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#8 /apps/web/dev3.site.com/public_html/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#9 /apps/web/dev3.site.com/public_html/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#10 /apps/web/dev3.site.com/public_html/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#11 /apps/web/dev3.site.com/public_html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#12 {main}

Next to the frontend error we face issues in the backend under content -> design configuration.
All theme values have become empty and It is not longer possible to set a theme as loading the edit page will not work. -> Load for ever.
I found that the database design_config_grid_flat table had the full name of the M1 theme after the data import. In M2 this needs to be a numerical value and so I changed it to the default 1 (blank or 2 luma). (Also the number of the categories source needs to be set in core_config_data.)
After the database change (string to numeric value) I was able to set the theme in the backend for one time only per store and after that the never ending loading occurred again.
The frontend error continues even after all these steps and clearing the cache numerous times.

After clearing all caches I found that I was still able to access two design edit pages. Saving the first page would give me the error: design_rule does not contain field "search"
Checking the second one that I could still open showed that the whole design rule section was gone/ empty and so giving me trouble saving anything.

**Comment by onepack** [comment]
Solved it by looking into the errors and matching the design exceptions (from the frontend error and the backend section that seemed to correlate to this as it was the design rules that was broken.
It had to be because of a database import as no core of design files else gets altered on a data migration.
So going through the database I came across an imported value that caused all these errors.
Whoever runs into this issue. Solve it by deleting in the core_config_data table the line with: the path
design/theme/skin_ua_regexp
and the value:
`a:1:{s:18:"_1335974008536_536";a:2:{s:6:"regexp";s:16:"Androidd|iPhoney";s:5:"value";s:6:"iphone";}}`
Deleting this line from the database solved the broken frondend and broken backend theme settings.

Hope this information helps anyone out there with the same issue!
